packages/nx/src/daemon/client/client.ts

```diff
@@ -380,16 +586,123 @@ export class DaemonClient {
         },
         (err) => callback(err, null)
       );
-      return messenger.sendMessage({
+      messenger.sendMessage({
         type: REGISTER_PROJECT_GRAPH_LISTENER,
       });
     });
 
+    // Return unregister function
     return () => {
-      messenger?.close();
+      this.projectGraphListenerCallbacks.delete(callbackId);
+
+      // If no more callbacks, close the connection
+      if (this.projectGraphListenerCallbacks.size === 0) {
+        this.projectGraphListenerMessenger?.close();
+        this.projectGraphListenerMessenger = undefined;
+      }
     };
   }
 
+  private async reconnectProjectGraphListener() {
```

Comment:
this also seems like dead code? 
It's somewhat inconsistent - the regular `registerProjectGraphRecomputationListener` dies when the daemon dies but this unused function handles reconnects properly. Maybe `reconnectProjectGraphListener` should be the new `registerProjectGraphRecomputationListener`?

Reply:
This is being used now.